website_sale_affiliate/models/sale_order.py

```
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    affiliate_request_id = fields.Many2one(
+        'sale.affiliate.request',
+        string='Affiliate request',
+        help='Affiliate request associated with sale order',
+    )
+
+    @api.model
+    def create(self, vals):
+        res = super(SaleOrder, self).create(vals)
+        affiliate = self.env['sale.affiliate'].find_from_session()
+        try:
+            res.affiliate_request_id = affiliate.get_request()
+        except AttributeError:
```

`AttributeError`?

---

`find_from_session` returns `None` in the event of no session, so attempting `get_request` would yield an `AttributeError` I believe

---

yes, "I believed" too :) But be explicit then and check if its `None`. Plus, if by error `get_request` goes on attr error you are going to swallow it without any good reason...

---

This goes against [EAFP](https://docs.python.org/2/glossary.html#term-eafp) though, which is pretty explicit in use case for keys and attributes.

---

bah, I just read 4 lines of code instead of one :stuck_out_tongue_closed_eyes:

---

> instead of one 

Whoah wait, one? In my head I see two (try, operation, except, pass) vs three (if, operation). Am I misunderstanding?

Also note that when v11 comes around, things like this will instead be replaced with:
```
with contextlib.suppress(AttributeError):
    operation()
```

Which would be the same two lines, but more Pythonic. Harder to identify what to switch during migrations if we switch to LBYL IMO

---

1 line :wink:
```python
res.affiliate_request_id = affiliate.get_request() if affiliate is not None else False
```

---

Yeah but what about our 79 char line limit?

---

Although I'll totally give you the fact that your's is cleaner if it weren't for that!

---

use backslash, break to new line and shut up! :smile: :clown_face: